docs/Evaluation.md

```diff
@@ -1,18 +1,86 @@
+
+## Mean-Average-Precision (mAP)
+
+mAP is the standard COCO evaluation metric and the most common for comparing computer vision models. It is useful as a summary statistic. However, it has several limitations for an ecological use case. 
+
+1. Not intuitive and difficult to translate to ecological applications. Read the sections above and visualize the mAP metric, which is essentially the area under the precision-recall curve at a range of IoU values. 
+2. The vast majority of biological applications use a fixed cutoff to determine an object of interest in an image. Perhaps in the future we will weight tree boxes by their confidence score, but currently we do things like, "All predictions > 0.4 score are considered positive detections". This does not connect well with the mAP metric.
+
+## Precision and Recall at a set IoU threshold.
+
+This was the original DeepForest metric, set a IoU = 0.4. This means that all predictions that overlap a ground truth box at IoU > 0.4 are true positives. As opposed to the torchmetrics above it is intuitive and matches downstream ecological tasks. The drawback is it slow, coarse, does not fully reward the model for having high confidence scores on true positives.
+
+There is an additional difference between ecological object detection like tree crowns and traditional computer vision methods. Instead of a single or set of easy to differentiate ground truths, we could have 60 or 70 objects that overlap in an image. How to best assign each prediction to each ground truth? DeepForest uses the [hungarian matching algorithm](https://thinkautonomous.medium.com/computer-vision-for-tracking-8220759eee85) to assign predictions to ground truth based on maximum IoU overlap. This is slow compared to the methods above, and so isn't a good choice for running hundreds of times during model training see config["validation"]["val_accuracy_interval"] for setting the frequency of the evaluate callback for this metric.
+
+# Calculating Evaluation Metrics 
+
+## Torchmetrics and loss scores
+
+These metrics are largely used during training to keep track of model performance. They are relatively fast and will be automatically run during training. 
+
+```    
+    m = main.deepforest()
+    csv_file = get_data("OSBS_029.csv")
+    root_dir = os.path.dirname(csv_file)
+    m.config["validation"]["csv_file"] = csv_file
+    m.config["validation"]["root_dir] = root_dir
+    results = m.trainer.validate(m)
+```
+This creates a dictionary of the average IoU ('iou') as well as 'iou' for each class. Here there is just one class, 'Tree'. Then the COCO mAP scores. See Further Reading above for explanation of mAP level scores. The val_bbox_regression is the loss function of the object detection box head, the loss_classification is the loss function of the object classification head.
```

Comment:
This creates a dictionary of the average IoU ('iou') as well as 'iou' for each class. Here there is just one class, 'Tree'. Then the COCO mAP scores. See Further Reading above for an explanation of mAP level scores. The val_bbox_regression is the loss function of the object detection box head, and the loss_classification is the loss function of the object classification head.

Reply:
Changed.